components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/GatewayArtifactsMgtDAO.java

```diff
@@ -384,11 +384,86 @@ public void deleteGatewayArtifacts(String apiId) throws APIManagementException {
         return apiRuntimeArtifactDtoList;
     }
 
+    public List<APIRuntimeArtifactDto> retrieveGatewayArtifactsByAPIIDs(List<String> apiIds, String[] labels,
+                                                                        String tenantDomain)
+            throws APIManagementException {
+        // Split apiId list into smaller list of size 25
+        List<List<String>> apiIdsChunk = new ArrayList<>();
+        int apiIdListSize = apiIds.size();
+        int apiIdArrayIndex = 0;
+        int apiIdsChunkSize = SQLConstants.API_ID_CHUNK_SIZE;
+        while (apiIdsChunkSize < apiIdListSize) {
+            apiIdsChunk.add(apiIds.subList(apiIdArrayIndex, apiIdsChunkSize));
+            apiIdListSize = apiIdListSize - SQLConstants.API_ID_CHUNK_SIZE;
+            apiIdArrayIndex = apiIdArrayIndex + SQLConstants.API_ID_CHUNK_SIZE;
+            apiIdsChunkSize = apiIdsChunkSize + SQLConstants.API_ID_CHUNK_SIZE;
+        }
+        if (apiIdListSize > 0) {
+            apiIdsChunk.add(apiIds.subList(apiIdArrayIndex, apiIdArrayIndex + apiIdListSize));
+        }
+        List<APIRuntimeArtifactDto> apiRuntimeArtifactDtoList = new ArrayList<>();
+
+        for (List<String> apiIdList: apiIdsChunk) {
+            apiRuntimeArtifactDtoList = new ArrayList<>(apiRuntimeArtifactDtoList);
```

Comment:
Is this line required? It seems the same `apiRuntimeArtifactDtoList` is getting initialized.

Reply:
Fixed with b47708f01abb761d5651b988a460491bb59d2e9d